lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -728,7 +728,10 @@ func (d *btrfs) MigrateVolume(vol Volume, conn io.ReadWriteCloser, volSrcArgs *m
 	// sendVolume sends a volume and its subvolumes (if negotiated subvolumes feature) to recipient.
 	sendVolume := func(v Volume, sourcePrefix string, parentPrefix string) error {
 		snapName := "" // Default to empty if volume isn't a snapshot and is main volume.
-		if v.IsSnapshot() {
+
+		// Detect snapshot by comparing to main volume.
+		// We can't use IsSnapshot() as the main vol may itself be a snapshot.
+		if v.name != vol.name {
```

Comment:
@stgraber I think we could slightly improve this here by adding an `&& v.IsSnapshot()` just for clarity really.